File: uilib/src/main/java/dev/entao/ui/util/XTextWatcher.java
package dev.entao.ui.util;

import android.text.Editable;

/**
 * Created by yet on 2015/10/15.
 */
public abstract class XTextWatcher implements android.text.TextWatcher {
    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {

    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {

    }

    @Override
    public void afterTextChanged(Editable s) {
        afterTextChanged(s.toString());
    }

    public abstract void afterTextChanged(String text);
}

File: uilib/src/main/java/dev/entao/kan/util/XTextWatcher.kt
package dev.entao.kan.util

import android.text.Editable

/**
 * Created by yet on 2015/10/15.
 */
abstract class XTextWatcher : android.text.TextWatcher {
    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

    }

    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

    }

    override fun afterTextChanged(s: Editable) {
        afterTextChanged(s.toString())
    }

    abstract fun afterTextChanged(text: String)
}
